src/fides/api/service/connectors/dynamic_erasure_email_connector.py

```diff
@@ -0,0 +1,502 @@
+                        privacy_request,
+                    )
+                )
+
+                # if we couldn't retrieve the email address / vendor, we skip the privacy request
+                # (error was already logged within the function, so we just continue to the
+                # next privacy request)
+                if not custom_field_data:
+                    continue
+
+                user_identities: Dict[str, Any] = (
+                    privacy_request.get_cached_identity_data()
+                )
+                filtered_user_identities: Dict[str, Any] = (
+                    filter_user_identities_for_connector(self.config, user_identities)
+                )
+
+                # If there are no user identities, we just skip the privacy request
+                if not filtered_user_identities:
+                    skipped_privacy_requests.append(privacy_request.id)
+                    self.add_skipped_log(db, privacy_request)
+                    continue
+
+                email, vendor_name = custom_field_data
+
+                key = f"{email}:{vendor_name}"
+                if key not in batched_identities:
+                    batched_identities[key] = BatchedIdentitiesData(
+                        email, vendor_name, []
+                    )
+
+                batched_identities[key].identities.extend(
+                    filtered_user_identities.values()
+                )
+
+            except Exception as exc:
+                logger.error(
+                    "An error occurred when retrieving email from custom request fields for connector {}. Skipping email send for privacy request with id {}. Error: {}",
+                    self.configuration.key,
+                    privacy_request.id,
+                    exc,
+                )
+                self.error_privacy_request(db, privacy_request, str(exc))
+
+        if not batched_identities:
+            logger.info(
+                "Skipping erasure email send for connector: '{}'. "
+                "No corresponding user identities or email addresses found for pending privacy requests.",
+                self.configuration.key,
+            )
+            return
+
+        logger.info(
+            "Sending batched erasure email for connector {}...",
+            self.configuration.key,
+        )
+
+        # Don't care about the keys in the dict, just the values
+        for batched_identity_data in batched_identities.values():
+            try:
+                send_single_erasure_email(
+                    db=db,
+                    subject_email=batched_identity_data.email_address,
+                    subject_name=batched_identity_data.vendor_name,
+                    batch_identities=batched_identity_data.identities,
+                    test_mode=False,
+                )
+            except MessageDispatchException as exc:
+                logger.info("Erasure email failed with exception {}", exc)
+                raise
```

Comment:
We talked about this I believe...but to check my understanding, here the email didn't send (say we had bad mailgund credentials), and all the associated privacy requests are left in a "awaiting email send" status to be picked up on the next email send. 

Would it be useful to add an execution log for every privacy request in this batch that the email send failed?

Reply:
That sounds reasonable. This part is copied from the Generic Erasure Email connector, which also doesn't add any execution logs for this case. But I can add them here if we don't find having slightly different behavior between both connectors

Reply:
Thanks @erosselli